### ISSUE TITLE
conditional transitions

### ISSUE THREAD

**Issue by ianstormtaylor** [anthonyshort/after-transition#3]
sorry this isn't really an issue, just a wish

the real use case i'm trying to solve in some of my components is being able to attach to a transition **if and only if** one happens. basic scenario is:

``` js
Modal.prototype.hide = function (callback) {
  this.emit('hiding');
  this.addClass('hide');
  var self = this;
  after(function () {
    self.emit('hide'); 
    callback && callback();
  });
  return this;
};

modal.hide(function () {
  console.log('hidden');
});
```

in an ideal world, i could style the modal however i want, adding `transition` properties if i so choose, so that the modal can **either** transition out or just disappear immediately.

then in javascript, i'd want the modal's `hide` event to fire when it is completely hidden. that means it would either fire after `0s` or potentially something like `.3s` if that's what the transition entailed.

basically, what i want is an `afterCurrentTransition` method.

i'm not sure if this is possible though, since the browser doesn't give us any way to know if a transition is currently happening. ideally, i'd add the hidden class, check if a transition is happening, and then decide what to do.

**Comment by ianstormtaylor** [comment]
just did way too much research, even though there are scant traces of `transitionstart` it's not in the spec yet... if that was there we could listen to all `transitionstart`s and mark a `.transitioning = true` property which would be hacky but solve the use case

**Comment by yields** [comment]
i think you are overcomplicating this haha, i just did [yields/after-transition], i have similiar use case just like yours and it works fine with just `.getComputedStyle(el).transition`.

maybe i didn't understand you correctly...

**Comment by ianstormtaylor** [comment]
nope you're right, im pretty sure i was just dumbly looking at the `.transition` div i was using in `.no-transition` div ;p

**Comment by ianstormtaylor** [comment]
as sorry, misunderstood. or maybe you responded in the wrong issue haha. for #2 you're right, i think just using `.transition` works.

for this issue though—ah you're right, wow i'm dumb. okay, it's not technically perfect though. for example, it could have a `transition: color .2s` and the check would fire, but the modal's `.hidden` class might only be changing `opacity`, in which case no transition happens, and the callback will never be called.

does that make sense? or am i missing another dumb thing.

**Comment by yields** [comment]
not sure i understood (english isn't my first language haha).
the callback will be called immediately if there's no transition, if there is it will wait for `transitionend` and call it.

this lib doesn't do this now, but it will @anthonyshort decides to use `.getComputedStyle()`.

sorry if i misunderstood again, haha :)

**Comment by anthonyshort** [comment]
I'll probably change this to use `getComputedStyle`, it makes way more sense.

**Comment by anthonyshort** [comment]
I've bumped the version to 0.0.4 and updated the `has-transitions` component 

Here's the changes: https://github.com/anthonyshort/has-transitions/blob/master/index.js

It uses the duration to determine if the element has any transitions at all. The browser will give elements a default transition value of "0s all 0s" or something, so we just need to check the duration. I'm using `getComputedStyle` here as well.

In some cases the duration property returns an empty string so this covers that too.

**Comment by anthonyshort** [comment]
[Here's a demo](https://rawgithub.com/anthonyshort/after-transition/master/demo/index.html)

**Comment by ianstormtaylor** [comment]
nice!

you could even just `!! parseFloat(duration)` if you wanted to i think and that would cover the empty string case. but i don't care either way :p
